gradle-plugins/rnbrownfield-gradle-plugin/rnbrownfield-plugin/src/main/kotlin/com/callstack/rnbrownfield/artifacts/ArtifactsResolver.kt

```diff
@@ -0,0 +1,138 @@
+@file:Suppress("DEPRECATION")
+
+/**
+ * Suppressing because of LibraryVariant.
+ * We can't use the new `com.android.build.gradle.api.LibraryVariant`
+ * as of now.
+ *
+ * We may want to re-visit this in future.
+ */
+
+package com.callstack.rnbrownfield.artifacts
+
+import com.android.build.gradle.LibraryExtension
+import com.android.build.gradle.api.LibraryVariant
+import com.callstack.rnbrownfield.plugin.ProjectConfigurations.Companion.CONFIG_NAME
+import com.callstack.rnbrownfield.plugin.ProjectConfigurations.Companion.CONFIG_SUFFIX
+import com.callstack.rnbrownfield.processors.VariantHelper
+import com.callstack.rnbrownfield.processors.VariantProcessor
+import com.callstack.rnbrownfield.processors.VariantTaskProvider
+import com.callstack.rnbrownfield.shared.BaseProject
+import com.callstack.rnbrownfield.shared.GradleProps
+import com.callstack.rnbrownfield.utils.Extension
+import org.gradle.api.ProjectConfigurationException
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.ResolvedArtifact
+import org.gradle.api.internal.artifacts.dependencies.DefaultProjectDependency
+
+class ArtifactsResolver(
+    private val configurations: MutableCollection<Configuration>,
+    private val baseProject: BaseProject,
+    private val extension: Extension,
+) :
+    GradleProps() {
+    companion object {
+        const val ARTIFACT_TYPE_AAR = "aar"
+        const val ARTIFACT_TYPE_JAR = "jar"
+    }
+
+    fun processArtifacts() {
+        embedDefaultDependencies("implementation")
+        setTransitiveToConfigurations()
+        generateArtifacts()
+    }
+
+    private fun setTransitiveToConfigurations() {
+        configurations.forEach {
+            if (baseProject.project.extensions.getByType(Extension::class.java).transitive) {
+                it.isTransitive = true
+            }
+        }
+    }
+
+    private fun embedDefaultDependencies(configName: String) {
+        val config = baseProject.project.configurations.findByName(configName)
+        val defaultDependencies = config?.dependencies?.filterIsInstance<DefaultProjectDependency>()
+        defaultDependencies?.forEach { dependency ->
+            if (extension.resolveLocalDependencies) {
```

Comment:
What is the purpose of this option? Would it make sense to make it on by default?